src/main/java/edu/umich/tl/analytics/CanvasCourseReportServlet.java

```diff
@@ -46,9 +51,12 @@
 	private static final String ACTION_GET_COURSES = "getCoursesPublished";
 	private static final String ACTION_GET_TERMS = "getEnrollmentTerms";
 	private static final String EMAIL_SUFFIX="@umich.edu";
+	private static final String ACTION_POLLING = "polling";
+	private static final String WORKING = "working";
 	private String canvasToken=CanvasCourseReportFilter.canvasToken;
 	private String canvasURL=CanvasCourseReportFilter.canvasURL;
 	ResourceBundle props = ResourceBundle.getBundle("coursereport");
+	ConcurrentHashMap<Thread,CourseReportTask> courseReportReqThreads=new ConcurrentHashMap<Thread,CourseReportTask>();
```

Comment:
Using a map in memory will work fine as long as all the requests go back to the same server.  If there are multiple servers and a load balancer that may or may not happen depending on configuration.  It's ok to do it this way as long if there aren't many servers and the load balancer is set up correctly.